### PR TITLE
docs: adding When to release or patch section

### LINKED ISSUE #236: doc: Explain when to create a new release vs a patch
A customer had this question [here](https://discord.com/channels/1030243211995791380/1247923530981445633) and I don't think we currently provide guidance.

### PR DESCRIPTION
**Pull request by erickzanardo**
## Status

**READY**

## Description

Adds some docs under the `Code push` section about when to choose for a release or when to choose a patch.

Open to suggestions on additions, removals, moving to a different section, etc.

Fixes #236

### CODE REVIEW

**Review by felangel — COMMENTED** [review]
I wonder if this should be added to FAQs instead of being it's own section. I think we could probably condense this quite a bit into a single FAQ:

Question: When should I create a patch?
Answer:
Generally, you should create patches when fixing bugs that DO NOT contain native changes or asset changes. For all other changes, you should likely create a new release. If distributing through stores, refer to store policies for additional guidelines.

**Comment by erickzanardo** [comment]
> I wonder if this should be added to FAQs instead of being it's own section. I think we could probably condense this quite a bit into a single FAQ:
> 
> Question: When should I create a patch? Answer: Generally, you should create patches when fixing bugs that DO NOT contain native changes or asset changes. For all other changes, you should likely create a new release. If distributing through stores, refer to store policies for additional guidelines.

@felangel that crossed my mind, but I feared that it would be lost among the lost list of the FAQ items that we currently have.

About the condensing part part, sounds good, I have this tendency of getting wordy sometimes 🙃

**Comment by felangel** [comment]
> > I wonder if this should be added to FAQs instead of being it's own section. I think we could probably condense this quite a bit into a single FAQ:
> > Question: When should I create a patch? Answer: Generally, you should create patches when fixing bugs that DO NOT contain native changes or asset changes. For all other changes, you should likely create a new release. If distributing through stores, refer to store policies for additional guidelines.
> 
> @felangel that crossed my mind, but I feared that it would be lost among the lost list of the FAQ items that we currently have.
> 
> About the condensing part part, sounds good, I have this tendency of getting wordy sometimes 🙃

I just think that this is a frequently asked question so it belongs in the FAQs. We have search built into our docs so I'm not worried about folks finding it 🤷

**Comment by bryanoltman** [comment]
It might also work as a subsection of "concepts" where we explain patches vs releases?

**Comment by felangel** [comment]
> It might also work as a subsection of "concepts" where we explain patches vs releases?

Yeah I would probably add this as an FAQ and link to it from the concepts sections on releases and patches.

**Comment by erickzanardo** [comment]
> > It might also work as a subsection of "concepts" where we explain patches vs releases?
> 
> Yeah I would probably add this as an FAQ and link to it from the concepts sections on releases and patches.

Roger Roger

File: src/content/docs/faq.mdx [hunk 1]
```diff
@@ -449,6 +449,13 @@ If you'd like to segment availability of Shorebird patches, there are 4 potentia
 
 Yes. Apps built with Shorebird will continue to function normally (as if they had been built without Shorebird), including those that have had patches installed.
 
+### When to send a patch
```
Comment (felangel) [inline src/content/docs/faq.mdx:452]:
```suggestion
### When to create a patch
```

File: src/content/docs/faq.mdx [hunk 2]
```diff
@@ -449,6 +449,13 @@ If you'd like to segment availability of Shorebird patches, there are 4 potentia
...
+
+Generally, you should send patches that DO NOT contain native changes or assets changes. For all other
+changes, you should likely create a new release.
+
+If distributing through the stores be sure to refer to their policies for additional guidelines.
```
Comment (felangel) [inline src/content/docs/faq.mdx:457]:
```suggestion
Generally, you should create patches when fixing bugs that **DO NOT** contain native changes or asset changes. For all other changes, you should likely create a new release.

If distributing through stores, please refer to store policies for additional guidelines.
```

File: src/content/docs/concepts.mdx [hunk 3]
```diff
@@ -70,6 +70,8 @@ When your application starts, it checks for available patches and applies the la
 
 Patches are created by running `shorebird patch [platform]`, where `platform` is `android`, `aar`, or `ios`.
 
+Check our [FAQ](/faq#when-to-send-a-patch) has some guiding on when to send a patch.
```
Comment (erickzanardo) [inline src/content/docs/concepts.mdx:73]:
```suggestion
Check our [FAQ](/faq#when-to-create-a-patch) has some guiding on when to send a patch.
```

Comment (felangel) [inline src/content/docs/concepts.mdx]:
```suggestion
:::note
For more information regarding when to create a patch, refer to our [FAQs](/faq#when-to-create-a-patch)
:::
```

File: src/content/docs/concepts.mdx [hunk 4]
```diff
@@ -70,6 +70,8 @@ When your application starts, it checks for available patches and applies the la
 
 Patches are created by running `shorebird patch [platform]`, where `platform` is `android`, `aar`, or `ios`.
 
+Check our [FAQ](/faq#when-to-create-a-patch) has some guiding on when to send a patch.
```
Comment (felangel) [inline src/content/docs/concepts.mdx:73]:
```suggestion
:::note
For more information regarding when to create a patch, refer to our [FAQs](/faq#when-to-create-a-patch)
:::
```

File: src/content/docs/faq.mdx [hunk 5]
```diff
@@ -449,6 +449,12 @@ If you'd like to segment availability of Shorebird patches, there are 4 potentia
 
 Yes. Apps built with Shorebird will continue to function normally (as if they had been built without Shorebird), including those that have had patches installed.
 
+### When to create a patch
```
Comment (felangel) [inline src/content/docs/faq.mdx:452]:
```suggestion
### When to create a patch?
```

File: src/content/docs/concepts.mdx [hunk 6]
```diff
@@ -70,6 +70,10 @@ When your application starts, it checks for available patches and applies the la
 
 Patches are created by running `shorebird patch [platform]`, where `platform` is `android`, `aar`, or `ios`.
 
+:::note
+For more information regarding when to create a patch, refer to our [FAQs](/faq#when-to-create-a-patch)
```
Comment (felangel) [inline src/content/docs/concepts.mdx:74]:
```suggestion
For more information regarding when to create a patch, refer to our [FAQs](/faq#when-should-i-create-a-patch)
```

**Review by felangel — APPROVED** [review]

**Review by bryanoltman — APPROVED** [review]

**Pull request merged**